tests/cluster_utils.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	restclient "k8s.io/client-go/rest"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	netclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+var kubeconfig *string
+var restConfig *restclient.Config
+var clientset *kubernetes.Clientset
+var netClient *netclient.K8sCniCncfIoV1Client
+const testNamespace = "test-namespace"
+
+func createTestNamespaces() error {
+	By(fmt.Sprintf("creating %s namespace", testNamespace))
+	_, err := clientset.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to create test namespace")
+	}
+	return nil
+}
+
+func deleteNamespace(namespace string) error {
+	return clientset.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+}
+
+func removeTestNamespaces() {
+	By(fmt.Sprintf("Waiting for namespace %s to be removed, this can take a while ...\n", testNamespace))
+	EventuallyWithOffset(1, func() bool { return apierrors.IsNotFound(deleteNamespace(testNamespace)) }, 120*time.Second, 5*time.Second).
+		Should(BeTrue(), "Namespace %s haven't been deleted within the given timeout", testNamespace)
+}
+
+func createPrivilegedPodWithIp(podName, nadName, cidr string) error {
+	By(fmt.Sprintf("Creating pod %s with priviliged premission and ip %s", podName, cidr))
+	privileged := new(bool)
+	*privileged = true
+	podObject := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{
+		Name: podName,
+		Namespace: testNamespace,
+		Annotations: map[string]string{"k8s.v1.cni.cncf.io/networks": nadName},
+	},
+	Spec: corev1.PodSpec{ Containers: []corev1.Container{{Name: "test",
+		Image:   "alpine",
+		Command: []string{"sh", "-c", fmt.Sprintf("ip address add %s dev net1; sleep 99999", cidr)},
```

Comment:
Isn't there a 'smarter' way to assign an IP address to the pod  on alpine ? We've stumbled on this particular rock plenty of times already.

Reply:
I'm afraid I don't know of such a way. can you direct me to somewhere it is done?

Reply:
Since we need to configure a secondary NIC and OVS does not support IPAM configuration, you have to do it "form inside the pod"
If it was we could configure it with calico https://docs.projectcalico.org/networking/use-specific-ip#how-to